File: devtools/project-core-extension-codestarts/src/main/resources/codestarts/quarkus/extension-codestarts/reactive-routes-codestart/java/src/test/java/org/acme/MyDeclarativeRoutesTest.java
package org.acme;

import static io.restassured.RestAssured.given;
import static org.hamcrest.CoreMatchers.is;

import org.junit.jupiter.api.Test;

import io.quarkus.test.junit.QuarkusTest;

@QuarkusTest
public class MyDeclarativeRoutesTest {

	@Test
	public void testHelloEndpoint() {
		given().when().get("/hello").then().statusCode(200).body(is("Hello RESTEasy Reactive Route"));
	}

	@Test
	public void testWorldEndpoint() {
		given().when().get("/world").then().statusCode(200).body(is("Hello world !!"));
	}

	@Test
	public void testGreetingEndpointWithNameParameter() {
		given().when().get("/greetings?name=Quarkus").then().statusCode(200).body(is("Hello  Quarkus !!"));
	}

	@Test
	public void testGreetingEndpointWithoutNameParameter() {
		given().when().get("/greetings").then().statusCode(200).body(is("Hello  RESTEasy Reactive Route !!"));
	}

}

File: devtools/project-core-extension-codestarts/src/main/resources/codestarts/quarkus/extension-codestarts/reactive-routes-codestart/kotlin/src/test/kotlin/org/acme/MyDeclarativeRoutesTest.kt
package org.acme

import io.quarkus.test.junit.QuarkusTest
import io.restassured.RestAssured.given
import org.hamcrest.CoreMatchers.`is`
import org.junit.jupiter.api.Test

@QuarkusTest
class ReactiveGreetingResourceTest {

    @Test
    fun testHelloEndpoint() {
        given()
          .`when`().get("/hello")
          .then()
             .statusCode(200)
             .body(`is`("Hello RESTEasy Reactive Route"))
    }

    @Test
    fun testWorldEndpoint() {
        given()
                .`when`().get("/world")
                .then()
                .statusCode(200)
                .body(`is`("Hello world !!"))
    }

    @Test
    fun testGreetingEndpointWithNameParameter() {
        given()
                .`when`().get("/greetings?name=Quarkus")
                .then()
                .statusCode(200)
                .body(`is`("Hello  Quarkus !!"))
    }

    @Test
    fun testGreetingEndpointWithoutNameParameter() {
        given()
                .`when`().get("/greetings")
                .then()
                .statusCode(200)
                .body(`is`("Hello  RESTEasy Reactive Route !!"))
    }
}